pkg/controller/jobframework/reconciler.go

```diff
@@ -622,7 +622,16 @@ func (r *JobReconciler) ensureOneWorkload(ctx context.Context, job GenericJob, o
 			return nil, nil
 		}
 
-		if err := r.ensurePrebuiltWorkloadOwnership(ctx, wl, object); err != nil {
+		if controllerutil.HasControllerReference(wl) && !metav1.IsControlledBy(wl, object) {
+			return nil, nil
+		}
```

Comment:
I remember the analogous check in https://github.com/kubernetes-sigs/kueue/pull/4069, and we added a warning log line. Should we do it here too?

Reply:
Good catch! It was missed during rebasing. Thanks!